src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java

```diff
@@ -173,13 +173,17 @@ Set<String> clientWarnings(KeyspacesDiff diff)
         return ImmutableSet.of();
     }
 
-    private void validateIndexTarget(TableMetadata table, IndexTarget target)
+    private void validateIndexTarget(TableMetadata table, IndexMetadata.Kind kind, IndexTarget target)
     {
         ColumnMetadata column = table.getColumn(target.column);
 
         if (null == column)
             throw ire("Column '%s' doesn't exist", target.column);
 
+        if ((kind == IndexMetadata.Kind.CUSTOM) && !SchemaConstants.isValidName(target.column.toString()))
+            throw ire("Column '%s' is longer than the permissible name length of %d characters or" +
+                      " contains non-alphanumeric-underscore characters", target.column, SchemaConstants.NAME_LENGTH);
```

Comment:
nit: Perhaps we could create a constant like `INVALID_CUSTOM_INDEX_TARGET_MESSAGE` w/ the format pattern here and then reuse it in the `shouldFailCreateWithInvalidCharactersInColumnName()` test.